File: src/test/java/no/nav/fo/veilarbregistrering/arbeidsforhold/ArbeidsforholdTest.java
package no.nav.fo.veilarbregistrering.arbeidsforhold;

import org.junit.jupiter.api.Test;

import java.time.LocalDate;

import static org.assertj.core.api.Assertions.assertThat;

public class ArbeidsforholdTest {

    @Test
    public void datoSkalVaereInneforPeriodeNaarTomErNull() {
        LocalDate mnd = LocalDate.of(2017,12,1);
        LocalDate fom = LocalDate.of(2010,12,1);
        Arbeidsforhold arbeidsforhold = ArbeidsforholdTestdataBuilder.medDato(fom, null);
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isTrue();
    }

    @Test
    public void datoSkalVaereInneforPeriode() {
        LocalDate mnd = LocalDate.of(2017,12,1);
        LocalDate fom = LocalDate.of(2017,12,1);
        LocalDate tom = LocalDate.of(2017,12,30);
        Arbeidsforhold arbeidsforhold = ArbeidsforholdTestdataBuilder.medDato(fom, tom);
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isTrue();
    }

    @Test
    public void datoSkalVaereInneforPeriode_2() {
        LocalDate mnd = LocalDate.of(2017,12,1);
        LocalDate fom = LocalDate.of(2017,10,1);
        LocalDate tom = LocalDate.of(2017,12,1);
        Arbeidsforhold arbeidsforhold = ArbeidsforholdTestdataBuilder.medDato(fom, tom);
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isTrue();
    }

    @Test
    public void datoSkalIkkeVaereInneforPeriode() {
        LocalDate mnd = LocalDate.of(2017,12,1);
        LocalDate fom = LocalDate.of(2017,9,1);
        LocalDate tom = LocalDate.of(2017,11,30);
        Arbeidsforhold arbeidsforhold = ArbeidsforholdTestdataBuilder.medDato(fom, tom);
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isFalse();
    }

}

File: src/test/java/no/nav/fo/veilarbregistrering/arbeidsforhold/ArbeidsforholdTest.kt
package no.nav.fo.veilarbregistrering.arbeidsforhold

import no.nav.fo.veilarbregistrering.arbeidsforhold.ArbeidsforholdTestdataBuilder.Companion.medDato
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDate

class ArbeidsforholdTest {
    @Test
    fun `dato skal være innefor periode naar tom er null`() {
        val mnd = LocalDate.of(2017, 12, 1)
        val fom = LocalDate.of(2010, 12, 1)
        val arbeidsforhold = medDato(fom, null)
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isTrue
    }

    @Test
    fun `dato skal være innefor periode`() {
        val mnd = LocalDate.of(2017, 12, 1)
        val fom = LocalDate.of(2017, 12, 1)
        val tom = LocalDate.of(2017, 12, 30)
        val arbeidsforhold = medDato(fom, tom)
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isTrue
    }

    @Test
    fun `dato skal være innefor periode 2`() {
        val mnd = LocalDate.of(2017, 12, 1)
        val fom = LocalDate.of(2017, 10, 1)
        val tom = LocalDate.of(2017, 12, 1)
        val arbeidsforhold = medDato(fom, tom)
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isTrue
    }

    @Test
    fun `dato skal ikke være innefor periode`() {
        val mnd = LocalDate.of(2017, 12, 1)
        val fom = LocalDate.of(2017, 9, 1)
        val tom = LocalDate.of(2017, 11, 30)
        val arbeidsforhold = medDato(fom, tom)
        assertThat(arbeidsforhold.erDatoInnenforPeriode(mnd)).isFalse
    }
}
